### PR TITLE
Support polling on multiple queues

### PR DESCRIPTION
**Pull request by alextwoods**
*Issue #, if available:*
#4 

*Description of changes:*
Adds support for running the poller on multiple queues.  You can either specify multiple queues in the arguments by passing `--queue` multiple times ( eg `--queue queue_1 --queue queue_2`) or for all configured queues by providing no queue arguments.

When multiple queues are specified, 1 thread per queue is started to run the poller for that queue.  All queues share a single, common thread pool for executing jobs.

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/aws/active_job/sqs/cli_options.rb [hunk 1]
```diff
@@ -90,8 +90,10 @@ def self.threads_option(opts, out)
           end
         end
 
-        def self.queue_option(opts, out)
-          opts.on('-q', '--queue STRING', '[Required] Queue to poll') { |a| out[:queue] = a }
+        def self.queues_option(opts, out)
+          opts.on('-q', '--queue STRING', 'Queue(s) to poll. You may specify this argument multiple times to poll multiple queues.  If not specified, will start pollers for all queues defined.') do |a|
```
Comment (mullermp) [inline lib/aws/active_job/sqs/cli_options.rb:94]: --queues with a comma separated string is a possibility

File: lib/aws/active_job/sqs/poller.rb [hunk 2]
```diff
@@ -54,30 +53,77 @@ def shutdown(timeout)
 
         def poll
           config = Aws::ActiveJob::SQS.config
-          queue = @options[:queue]
+          if @queues && !@queues.empty?
+            if @queues.size == 1
+              # single queue, use main thread
```
Comment (mullermp) [inline lib/aws/active_job/sqs/poller.rb:58]: Why the distinction? Could always use a background thread.

File: lib/aws/active_job/sqs/poller.rb [hunk 3]
```diff
@@ -54,30 +53,77 @@ def shutdown(timeout)
...
+              @logger.info "Background Polling on: #{queue} => #{queue_url} with options=#{poller_options}"
+
+              _poll(poller_options, queue_url)
+            end
+          end
+          poller_threads.each(&:join)
+        end
+
+        def validate_config(queue)
+          return if Aws::ActiveJob::SQS.config.queues[queue.to_sym]&.fetch(:url, nil)
```
Comment (mullermp) [inline lib/aws/active_job/sqs/poller.rb:98]: Perhaps queues can already have symbol names when you parse the option

File: lib/aws/active_job/sqs/cli_options.rb (hunk 1 above)
Comment (alextwoods) [reply lib/aws/active_job/sqs/cli_options.rb:94]: I prefer multiple `--queue` options personally and I think is often more the standard way to provide multiple arguments.  Eg: `ruby`/`irb` allows multiple `-r` flags, eg `irb -I./ -r req1.rb -r req2.rb`

File: lib/aws/active_job/sqs/poller.rb (hunk 2 above)
Comment (alextwoods) [reply lib/aws/active_job/sqs/poller.rb:58]: Less resource usage - you shouldn't spawn an additional thread that we don't need.  And I want to keep encouraging this as the preferred approach.

File: README.md [hunk 4]
```diff
@@ -196,8 +196,17 @@ for documentation.
 
 To start processing jobs, you need to start a separate process
 (in additional to your Rails app) with `bin/aws_active_job_sqs`
-(an executable script provided with this gem).  You need to specify the queue to
-process jobs from:
+(an executable script provided with this gem). You can poll for one or multiple
+queues using the `--queue` argument(s). You can  specify multiple queues
```
Comment (jterapin) [inline README.md:200]: nit: double spacing between "can" and "specify"

Comment (jterapin) [reply README.md:200]: Also could start a new line on multiple queues for ease of readability (personal preference, however)

**Review by mullermp — APPROVED** [review]

**Pull request merged**